core/templates/pages/learner-dashboard-page/content-toggle-button/content-toggle-button.component.html

```diff
@@ -0,0 +1,30 @@
+<button class="align-self-end content-toggle-button mt-4" (click)="toggle()">
+  <span class="content-toggle-button-text"> {{ buttonText }} </span>
+  <span class="fas" [ngClass]="isExpanded ? 'fa-chevron-up' : 'fa-chevron-down'"></span>
+</button>
+
+<style>
+  :host {
```

Comment:
Ditto as above, which class are we targeting with the host tag? And why shouldn't the style be defined in the host element itself?

Reply:
Same as [above](https://github.com/oppia/oppia/pull/20929#discussion_r1741053205). Removing fit-content would have <oppia-content-toggle-button></oppia-content-toggle-button> expand to parent's width. Moving align-self: end to .content-toggle-button won't work since the parent only inherits d-flex, not flex-column. I would declare justify-content: end in :host instead to achieve the same effect.

Reply:
Can't we just create an outer element for the button component inside the parent components to handle the styles? These styles don't belong at the component level for a button and I think we should usually avoid that.

Reply:
I messed around with the classes and was able to implement the styling without using :host. Done.